modules/apps/item-selector/item-selector-taglib/src/main/resources/META-INF/resources/item_selector_preview/js/ItemSelectorPreviewHook.es.js

```diff
@@ -0,0 +1,263 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {useIsMounted} from 'frontend-js-react-web';
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+import ReactDOM from 'react-dom';
+
+import Carousel from './Carousel.es';
+import Footer from './Footer.es';
+import Header from './Header.es';
+
+const KEY_CODE = {
+	ESC: 27,
+	LEFT: 37,
+	RIGTH: 39
+};
+
+const TPL_EDIT_DIALOG_TITLE = '{edit} {title} ({copy})';
+
+const ItemSelectorPreview = ({
+	container,
+	currentIndex = 0,
+	editItemURL,
+	handleSelectedItem,
+	headerTitle,
+	items,
+	uploadItemReturnType,
+	uploadItemURL
+}) => {
+	const [currentItem, setCurrentItem] = useState(items[currentIndex]);
+	const [currentItemIndex, setCurrentItemIndex] = useState(currentIndex);
+	const [itemList, setItemList] = useState(items);
+
+	const infoButtonRef = React.createRef();
+
+	const isMounted = useIsMounted();
+
+	useEffect(() => {
+		document.documentElement.addEventListener('keydown', handleOnKeyDown);
+
+		const sidenavToggle = infoButtonRef.current;
+
+		if (sidenavToggle) {
+			Liferay.SideNavigation.initialize(sidenavToggle, {
+				container: '.sidenav-container',
+				position: 'right',
+				typeMobile: 'fixed',
+				width: '320px'
+			});
+		}
+
+		const updateCurrentItemHandler = Liferay.on(
+			'updateCurrentItem',
+			updateCurrentItem
+		);
+
+		return () => {
+			document.documentElement.removeEventListener(
+				'keydown',
+				handleOnKeyDown
+			);
+
+			Liferay.detach(updateCurrentItemHandler);
+		};
+	}, [handleOnKeyDown, infoButtonRef, updateCurrentItem]);
+
+	const close = () => {
+		ReactDOM.unmountComponentAtNode(container);
+	};
+
+	const handleClickDone = () => {
+		handleSelectedItem(currentItem);
+		close();
+	};
+
+	const handleClickEdit = () => {
+		const itemTitle = currentItem.title;
+
+		const editDialogTitle = Liferay.Util.sub(TPL_EDIT_DIALOG_TITLE, {
```

Comment:
Maybe this works, or maybe I forget something:

```js
const editDialogTitle = `${
  Liferay.Language.get('copy')
} ${
  Liferay.Language.get('edit')
} (${itemTitle})`;
```

Reply:
Sounds good